webapp/loggedin/modules/analysis/processingChain/components/processingChainStep.js

```diff
@@ -18,9 +20,15 @@ import ErrorBadge from '@webapp/commonComponents/errorBadge'
 const ProcessingChainStep = props => {
   const { processingStep, validation } = props
 
-  const entityUuid = ProcessingStep.getEntityUuid(processingStep)
   const survey = useSelector(SurveyState.getSurvey)
-  const entity = entityUuid ? Survey.getNodeDefByUuid(entityUuid)(survey) : null
+  const entity = R.pipe(
+    ProcessingStep.getEntityUuid,
+    R.unless(R.isNil, entityUuid => Survey.getNodeDefByUuid(entityUuid)(survey)),
```

Comment:
You might not need R.unless. Survey.getThingByUuid will be return null if the uuid you pass as parameter is null

Reply:
right, but it avoids a useless overhead, like getting the list of nodes object and trying to get a "null" property of it...

Reply:
You don't get the list of nodes.

Reply:
yes, you don't traverse a list but still you get a shallow copy of the nodes object and then you try to get a property "null" of it.
But since it will make the code more readable, I'll remove it.

Reply:
then the R.unless should be added to ```Survey.getThingByUuid```. what do you think?